client/jetpack-connect/signup.js

```diff
@@ -83,11 +95,49 @@ export class JetpackSignup extends Component {
 		} );
 	}
 
+	componentDidUpdate( prevProps ) {
+		const { requestError } = this.props;
+
+		if ( prevProps.requestError || ! requestError ) {
+			return;
+		}
+
+		if (
+			this.getWooDnaConfig() &&
+			'usernameOrEmail' === requestError.field &&
+			'unknown_user' === requestError.code
+		) {
+			this.showWooDnaSignupView();
+		}
+	}
+
+	showWooDnaSignupView = () => {
+		this.setState( {
+			wooDnaFormType: 'signup',
```

Comment:
There's no need, [these other state values](https://github.com/Automattic/wp-calypso/blob/1aebc79bc32e3de59174e88c633964a58dfe1c6f/client/jetpack-connect/signup.js#L125-L126) are only used in the login view, so there's no need to clear them here. And we definitely don't want to clear the `signUpUsernameOrEmail` value, since that will be use to prefill the signup form.